### PR TITLE
[14.0][MIG] sale_promotion_rule_delivery

### PR DESCRIPTION
**Pull request by xavier-bouquiaux**
some change in test
delivery_set(self) does not exists anymore
wizard choose.delivery.carrier used instead

### CODE REVIEW

**Comment by rousseldenis** [comment]
Depends on:

- [ ] #1583 

Remove:

- [ ] test-requirements.txt

**Comment by xavier-bouquiaux** [comment]
rebased as #1583  has been merged

**Comment by xavier-bouquiaux** [comment]
@rousseldenis , @sebastienbeau 
as we merged #1583 we could merge this ?

What do you think ?

**Review by rousseldenis — APPROVED** [review]

File: sale_promotion_rule_delivery/models/sale_promotion_rule.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+from odoo import models
+
+
+class SalePromotionRule(models.Model):
+
+    _inherit = "sale.promotion.rule"
+
+    def _get_lines_excluded_from_total_amount(self, order):
+        lines = super(SalePromotionRule, self)._get_lines_excluded_from_total_amount(
```
Comment (acsonefho) [inline sale_promotion_rule_delivery/models/sale_promotion_rule.py]: Python3 please `super()._get_lines_excluded_from_total_amount(order)`

File: sale_promotion_rule_delivery/models/sale_promotion_rule.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+            order
+        )
+        lines |= order.order_line.filtered("is_delivery")
+        return lines
+
+    def _is_promotion_valid_for_line(self, line):
+        self.ensure_one()
+        if line.is_delivery:
+            return False
+        return super(SalePromotionRule, self)._is_promotion_valid_for_line(line)
```
Comment (acsonefho) [inline sale_promotion_rule_delivery/models/sale_promotion_rule.py]: Same here: `super()._is_promotion_valid_for_line(line)`

File: sale_promotion_rule_delivery/i18n/ca.po [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+"MIME-Version: 1.0\n"
+"Content-Type: text/plain; charset=UTF-8\n"
+"Content-Transfer-Encoding: \n"
+"Plural-Forms: nplurals=2; plural=n != 1;\n"
+"X-Generator: Weblate 4.3.2\n"
+
+#. module: sale_promotion_rule_delivery
+#: model:ir.model,name:sale_promotion_rule_delivery.model_sale_promotion_rule
+msgid "Sale Promotion Rule"
+msgstr "Regla de promoció de la venda"
```
Comment (acsonefho) [inline sale_promotion_rule_delivery/i18n/ca.po:22]: It's spanish. This file should be updated or removed.

File: sale_promotion_rule_delivery/tests/test_promotion.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+from odoo.tests.common import Form, TransactionCase
+
+from odoo.addons.sale_promotion_rule.tests.test_promotion import (
+    AbstractCommonPromotionCase,
+)
+
+
+class PromotionCase(TransactionCase, AbstractCommonPromotionCase):
+    def setUp(self, *args, **kwargs):
+        super(PromotionCase, self).setUp(*args, **kwargs)
```
Comment (acsonefho) [inline sale_promotion_rule_delivery/tests/test_promotion.py]: `super().setUp(*args, **kwargs)`

File: sale_promotion_rule_delivery/tests/test_promotion.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+                {
+                    "default_order_id": self.sale.id,
+                    "default_carrier_id": self.sale.carrier_id,
+                }
+            )
+        )
+        choose_delivery_carrier = delivery_wizard.save()
+        choose_delivery_carrier.update_price()
+        choose_delivery_carrier.button_confirm()
+        # self.sale.delivery_set()
```
Comment (acsonefho) [inline sale_promotion_rule_delivery/tests/test_promotion.py]: Remove

**Review by acsonefho — CHANGES_REQUESTED** [review]

File: sale_promotion_rule_delivery/i18n/ca.po (hunk 3 above)
Comment (rousseldenis) [reply sale_promotion_rule_delivery/i18n/ca.po:22]: @acsonefho No, it's catalan.

**Comment by rousseldenis** [comment]
@xavier-bouquiaux could you attend @acsonefho comments?

**Comment by xavier-bouquiaux** [comment]
> @xavier-bouquiaux could you attend @acsonefho comments?

Done

**Review by acsonefho — APPROVED** [review]

**Comment by xavier-bouquiaux** [comment]
@rousseldenis : let's merge this ?

**Comment by rousseldenis** [comment]
@xavier-bouquiaux Could you squash those commits ? 
image

**Comment by acsonefho** [comment]
@xavier-bouquiaux Can you rebase and squash please? :pray: 

> @xavier-bouquiaux Could you squash those commits ? image

**Comment by xavier-bouquiaux** [comment]
@rousseldenis @acsonefho 
could we merge this ?

**Pull request merged**